src/partition/FileUtils.h

```diff
@@ -46,8 +48,12 @@ class InputFile {
     /// Return the path of the input file.
     boost::filesystem::path const &path() const { return _path; }
 
+    // Needed in derived class InputFileArrow
+    virtual int getBatchNumber() const { return -1; }
+
     /// Read a range of bytes into `buf`.
-    void read(void *buf, off_t off, size_t sz) const;
+    virtual void read(void *buf, off_t off, size_t sz, int *bufferSize = 0,
+                      std::vector<std::string> params = {}) const;
```

Comment:
If you add virtual methods, destructor also needs to be made virtual.

Reply:
And for the sake of efficiency, the last parameter in the method's signature needs to be defined as:
```
std::vector<std::string> const& params = std::vector<std::string>
```

Reply:
Also the following parameter:
```
int *bufferSize = 0
```
Should be redeclared to be a reference:
```
int& bufferSize,
```
This would simplify the rest of your code.

Reply:
Done